api/resources_portal/management/commands/material_additional_metadata_key_change.py

```diff
@@ -0,0 +1,183 @@
+from django.core.management.base import BaseCommand, CommandError
+from django.utils import timezone
+
+from resources_portal.config.logging import get_and_configure_logger
+from resources_portal.models import Material
+
+logger = get_and_configure_logger(__name__)
+
+
+def validate_category(category):
+    return len([choice for choice in Material.CATEGORY_CHOICES if category in choice]) > 0
+
+
+def paginate_queryset(query_set, function, pagination):
+    current_page = 0
+    size = len(query_set)
+    current_slice = []
+    materials_updated = 0
+
+    while current_page < size:
+        if (size - current_page) < pagination:
+            current_slice = query_set[current_page : current_page + (size - current_page)]
+            current_page += size - current_page
+        else:
+            current_slice = query_set[current_page : current_page + pagination]
+            current_page += pagination
+
+        current_materials = []
+        for material in current_slice:
+            current_material = function(material)
+            current_material.updated_at = timezone.now()
+            current_materials.append(current_material)
```

Comment:
I don't think you need to create a new list here. You can just pass the queryset to `bulk_update` and it doesn't look like you need to keep track of the return value of `function` because they just update the object that gets passed into them.

Reply:
I tested this out, and I'm not sure if it's accurate. The `function` just modifies that material object currently loaded into memory, whereas the queryset is reflective of what's in the database, not in memory. I don't think it's especially computationally expensive to keep track of the data of a single slice of the materials in RAM at a time.

Reply:
Ok, works for me.